newjersey/navigator.business.nj.gov
# feat: [#183443674] update c-corp formation

NOTE: Waiting on feedback from the PrepareFiling API team. Currently, cannot successfully submit. Getting the following error:

```
[{"Valid":false,"ErrorMessage":"Required","Name":"Formation.ForeignGoodStandingFile"}]
```

---

## Description

Updates the flow for foreign c-corp formation.

### Ticket

[183443674](https://www.pivotaltracker.com/story/show/183443674)

### Approach

- Create new components associated with new formation questions
  - `PracticesLaw`
  - `ForeignCertificate`
- Create a new `FileInput` component using the USWDS component and functionality as a launchpad
- Established new types and updated config files as necessary
- Updated existing tests to be more flexible and to accommodate for the new `foreignGoodStandingFile` that will be passed around alongside formation data

We are not storing the `foreignGoodStandingFile` but it needs to be maintained in state while the user is filling out the form prior to submission. To accomplish this, we created a new state object. Passing this around required updating a number of functions to accept a props object (easier to handle optional props) and modifying some types.

### Steps to Test

You can experiment with this work by creating a foreign c-corp. The UI changes should be on the formation Business step and Review step. Clicking submit should hit the NICUSA API. Check for errors and successful submission to NICUSA.

### Notes

n/a

## Code author checklist

  - This does in fact use a relative import to get the USWDS library to play nicely.
